src/storm/solver/TopologicalMinMaxLinearEquationSolver.cpp

```diff
@@ -240,36 +271,53 @@ namespace storm {
         
         template<typename ValueType>
         bool TopologicalMinMaxLinearEquationSolver<ValueType>::solveScc(storm::Environment const& sccSolverEnvironment, OptimizationDirection dir, storm::storage::BitVector const& sccRowGroups, storm::storage::BitVector const& sccRows, std::vector<ValueType>& globalX, std::vector<ValueType> const& globalB) const {
-
             // Set up the SCC solver
             if (!this->sccSolver) {
                 this->sccSolver = GeneralMinMaxLinearEquationSolverFactory<ValueType>().create(sccSolverEnvironment);
                 this->sccSolver->setCachingEnabled(true);
             }
-            if (this->choiceFixedForState) {
-                // convert fixed states to only fixed states of sccs
+            this->sccSolver->setHasUniqueSolution(this->hasUniqueSolution());
+            this->sccSolver->setHasNoEndComponents(this->hasNoEndComponents());
+            this->sccSolver->setTrackScheduler(this->isTrackSchedulerSet());
+
+            storm::storage::SparseMatrix<ValueType> sccA;
+            if (this->choiceFixedForRowGroup) {
+                // Obtain choiceFixedForState bitvector containing only the states of the scc.
                 storm::storage::BitVector choiceFixedForStateSCC(sccRowGroups.getNumberOfSetBits());
```

Comment:
This is shorter compared to the computation below:
```suggestion
                storm::storage::BitVector choiceFixedForStateSCC = this->choicesFixedForRowGroup % sccRowGroups;
```
Although I think `choiceFixedForStateSCC` is not used? Maybe there is a typo below.